test/integration/etcd/client_test.go

```diff
@@ -0,0 +1,304 @@
+		})
+		Expect(err).ShouldNot(HaveOccurred())
+		defer adminClient.Close()
+		adminClient.RoleAdd(context.Background(), "root")
+		adminClient.UserAdd(context.Background(), "root", "root")
+		adminClient.UserGrantRole(context.Background(), "root", "root")
+		adminClient.UserAdd(context.Background(), username, password)
+		adminClient.UserGrantRole(context.Background(), username, "root")
+		adminClient.AuthEnable(context.Background())
+
+		ports, err := test.AllocateFreePorts(2)
+		Expect(err).NotTo(HaveOccurred())
+		addr := fmt.Sprintf("%s:%d", host, ports[0])
+		httpAddr := fmt.Sprintf("%s:%d", host, ports[1])
+		closeFn := setup.CMD("liaison",
+			"--namespace", namespace,
+			"--grpc-host="+host,
+			fmt.Sprintf("--grpc-port=%d", ports[0]),
+			"--http-host="+host,
+			fmt.Sprintf("--http-port=%d", ports[1]),
+			"--http-grpc-addr="+addr,
+			"--node-host-provider", "flag",
+			"--node-host", nodeHost,
+			"--etcd-endpoints", etcdEndpoint,
+			"--etcd-username", username,
+			"--etcd-password", password)
+		defer closeFn()
+
+		Eventually(helpers.HTTPHealthCheck(httpAddr), flags.EventuallyTimeout).Should(Succeed())
+		Eventually(func() (map[string]*databasev1.Node, error) {
+			return listKeys(etcdEndpoint, username, password, clientConfig, fmt.Sprintf("/%s/nodes/%s:%d", namespace, nodeHost, ports[0]))
+		}, flags.EventuallyTimeout).Should(HaveLen(1))
+	})
+
+	It("should be using cacert connect etcd server successfully", func() {
+		serverTLSInfo := transport.TLSInfo{
+			KeyFile:  serverKeyFilePath,
+			CertFile: serverCertFilePath,
+		}
+		clientTLSInfo := transport.TLSInfo{
+			TrustedCAFile: caFilePath,
+		}
+		clientConfig, err := clientTLSInfo.ClientConfig()
+		Expect(err).ShouldNot(HaveOccurred())
+
+		etcdServer, err := createEtcdServer(dir, serverTLSInfo)
+		Expect(err).ShouldNot(HaveOccurred())
+		// wait for e.Server to join the cluster
+		<-etcdServer.Server.ReadyNotify()
+		etcdEndpoint := etcdServer.Config().ListenClientUrls[0].String()
+		defer etcdServer.Close()
+
+		ports, err := test.AllocateFreePorts(2)
+		Expect(err).NotTo(HaveOccurred())
+		addr := fmt.Sprintf("%s:%d", host, ports[0])
+		httpAddr := fmt.Sprintf("%s:%d", host, ports[1])
+		closeFn := setup.CMD("liaison",
+			"--namespace", namespace,
+			"--grpc-host="+host,
+			fmt.Sprintf("--grpc-port=%d", ports[0]),
+			"--http-host="+host,
+			fmt.Sprintf("--http-port=%d", ports[1]),
+			"--http-grpc-addr="+addr,
+			"--node-host-provider", "flag",
+			"--node-host", nodeHost,
+			"--etcd-endpoints", etcdEndpoint,
+			"--etcd-tls-ca-file", caFilePath)
+		defer closeFn()
+
+		Eventually(helpers.HTTPHealthCheck(httpAddr), flags.EventuallyTimeout).Should(Succeed())
+		Eventually(func() (map[string]*databasev1.Node, error) {
+			return listKeys(etcdEndpoint, "", "", clientConfig, fmt.Sprintf("/%s/nodes/%s:%d", namespace, nodeHost, ports[0]))
+		}, flags.EventuallyTimeout).Should(HaveLen(1))
+	})
+
+	It("should be using key pair connect etcd server successfully", func() {
```

Comment:
As we discussed earlier, the procedure is not working correctly, as it cannot connect with the server. 

How do you fix it?